complaints/static/main.js

```diff
@@ -5,81 +5,103 @@ function urg_fun() {
-		string = "http://" + string;
-	}
-	abc.value = string
-}
-
-function timerecord1() {
-	var t1 = document.getElementById('time').value;
-	if (t1.length == 5) { document.getElementById('hiddentime').value = t1 + ':00'; }
-	else { document.getElementById('hiddentime').value = t1; }
-	y = document.getElementById('hiddentime2').value;
-	x = document.getElementById('hiddentime').value;
-	var y = new Date("01/01/2007 " + y);
-	var x = new Date("01/01/2007 " + x);
-	var x = y - x;
-	x = x / 60 / 60 / 1000;
-	if (x < 1) { document.getElementById('time_alert').style.display = ""; document.getElementById('b1').setAttribute('disabled', ''); }
-	else { document.getElementById('time_alert').style.display = "none"; document.getElementById('b1').removeAttribute('disabled'); }
-};
-function timerecord2() {
-	var t2 = document.getElementById('time2').value;
-	if (t2.length == 5) { document.getElementById('hiddentime2').value = t2 + ':00'; }
-	else { document.getElementById('hiddentime2').value = t2; }
-	y = document.getElementById('hiddentime2').value;
-	x = document.getElementById('hiddentime').value;
-	var y = new Date("01/01/2007 " + y);
-	var x = new Date("01/01/2007 " + x);
-	var x = y - x;
-	x = x / 60 / 60 / 1000;
-	if (x < 1) { document.getElementById('time_alert').style.display = ""; document.getElementById('b1').setAttribute('disabled', ''); }
-	else { document.getElementById('time_alert').style.display = "none"; document.getElementById('b1').removeAttribute('disabled'); }
-};
-
-function comp_fun(id) {
-	if (document.getElementById(id + 'a').style.display == "none") {
+	function timerecord1() {
+		var t1 = document.getElementById('time').value;
+		if(t1.length == 5){document.getElementById('hiddentime').value = t1 + ':00';}
+		else{document.getElementById('hiddentime').value = t1;}		
+		y = document.getElementById('hiddentime2').value;
+		x = document.getElementById('hiddentime').value;
+        var y = new Date("01/01/2007 " + y);
+        var x = new Date("01/01/2007 " + x);
+		var x = y - x;
+		x = x / 60 / 60 / 1000;
+		if(x < 1){document.getElementById('time_alert').style.display = "";document.getElementById('b1').setAttribute('disabled','');}
+		else{document.getElementById('time_alert').style.display = "none";document.getElementById('b1').removeAttribute('disabled');}
+		var today = new Date();
+		p = (today.getMonth()+1) + '/' + today.getDate() + '/' + today.getFullYear();
+		console.log(p);
+		var timenow = new Date(p + " " + today.getHours() + ":" + today.getMinutes() + ":" + today.getSeconds());
+		var start_time = new Date(p + " " + document.getElementById('hiddentime').value);
+		var staff_start = new Date(p + " " + "09:00:00");
+		console.log("now: " + timenow);
+		console.log("entered: " + start_time);
+		//if (start_time <= timenow){console.log("Working");}
+		if(start_time <= timenow || start_time < staff_start){document.getElementById('time_alert_x').style.display = "";document.getElementById('b1').setAttribute('disabled','');}
```

Comment:
Will this surely work properly?Because before checking you are appending today's date in each time.So if person has selected tomorrow's date with time less than timenow then it will show an error although it is a completely valid input.